### ISSUE TITLE
negative emotions summary

### ISSUE THREAD

**Issue by wookietreiber** [DFHack/dfhack#960]
I have recently been thinking about a tool that would display a summary of negative emotions, i.e. those that increase stress. This way, you can easily find out what your fortress is generally lacking and improve overall happiness by satisfying many at once.

### Example Output

I think its enough if this gets printed directly to the console, like prospect does, there does not necessarily need to be a more involved user interface.

```
...
90% (180) ...
95% (190) annoyed after eating without a table.
100% (200) annoyed after having a drink without using a goblet.
```

The output shows the percentage of the population as well as the exact amount and the negative emotion. They are sorted in increasing order.

### Pseudo Code

```
badEmotions = foreach citizen get emotions filter bad ones
foreach badEmotion.text dropPronoun | sort | uniq -c
```

### Other Thoughts

- maybe add some command line options like `bad-emotions -10` to show only the highest ten emotions instead of all by default
- could also be done for needs, but I think that emotions and needs should be output separately

**Comment by PeridexisErrant** [comment]
I believe that [dwarfmonitor stats](http://dfhack.readthedocs.io/en/stable/docs/Plugins.html#dwarfmonitor) does this already, though it could certainly be better documented.

(it's not clear from the docs what this command actually does, which certainly deserves an open issue until fixed.)

**Comment by lethosor** [comment]
I don't think it displays specific thoughts, though.

**Comment by lethosor** [comment]
No, this is `dwarfmonitor stats`:

![image](https://cloud.githubusercontent.com/assets/3719547/16778395/d5afe40c-483b-11e6-9d52-b60192420a2d.png)

**Comment by PeridexisErrant** [comment]
Ah, my mistake then - I must have misremembered the preferences option.  `dwarfmonitor` is probably the logical place for this functionality though.

**Comment by wookietreiber** [comment]
Yes, indeed, dwarfmonitor looks like the most logical tool to extend.

What are the chances this gets implemented soon? How can I help to speed this up (apart from providing a PR)?

**Comment by lethosor** [comment]
Most of our effort right now is going into 0.43.05/64-bit/VS2015 support, so most new features will probably take a while. I guess you could help test any pre-releases once those come out to help us speed up the process and get back to work on features, but we're not really there yet.
